### ISSUE TITLE
Inline link contrast - inconsistency depending on the text content of the link.

### ISSUE THREAD

**Issue by Sereza7** [dequelabs/axe-core#4314]
### Product

axe-core

### Product Version

4.8.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
The product reported both commentor names in the same way, either passed or failed.
#### Actual
One of the commentor name is reported (name = 123), while the other is not (name = Christopher W.).

#### How to Reproduce
This is the original situation where I found out this inconsistency (working on xwiki-platform accessibility):
![InlineLinkDetection2](https://github.com/dequelabs/axe-core/assets/28761965/2af0de55-766e-40e3-9543-31564ce9bc25)
![InlineLinkDetection1](https://github.com/dequelabs/axe-core/assets/28761965/7f4cf586-b6ee-469d-a451-84aa7986ae4b)

#### Additional context
The bug here is the inconsistency.

**Comment by straker** [comment]
Thanks for the issue. Could you provide a way to reproduce the issue, either a website that shows the issue or some code? I tried a quick test with what you provided but in the following example both links show violations for the `link-in-text-block` rule.

```html
<p>
  <span><a style="text-decoration: none"; href="#">123</a></span>, <span>2024/12/12</span> <span>13:45</span></p>
</p>
<p>
  <span><a style="text-decoration: none"; href="#">John Doe</a></span>, <span>2024/12/12</span> <span>13:45</span></p>
</p>
```

**Comment by Sereza7** [comment]
Thanks for your answer @straker!
I tried to reproduce with a basic example too, but couldn't figure out how :/

Here is an archive of the exported HTML (with related files).
Unfortunately there's for sure a lot of context that's not relevant to the issue at hand but at least that's a way to reproduce the issue.
[Sandbox - XWiki.zip](https://github.com/dequelabs/axe-core/files/14109704/Sandbox.-.XWiki.zip)

**Comment by straker** [comment]
Unfortunately, for security reasons we cannot download / open a zip/archive file. Is there a different way you could send the information?

**Comment by Sereza7** [comment]
Sorry, it's not much better security-wise, but I uploaded the HTML and sources in [a GitHub repository](https://github.com/Sereza7/HTMLexample-Inline-link-contrast---inconsistency-depending-on-the-text-content-of-the-link).

**Comment by straker** [comment]
Thanks. I can take a look later today and see if I can figure out the problem

**Comment by WilcoFiers** [comment]
@Sereza7 We had a look at this. The reason this isn't reported is because there is more link than other text. This rule has a few heuristics in place to avoid false positives. Comparing the link text to the text of the. paragraph is one of them. So this is expected behavior. I'm not sure there is anything we can do about this without increasing the risk of this rule reporting false positives. But this definitely is an issue.
